Editoria11y Tests for Headings

emerham opened the pull request:
Adds tests to Editoria11y for Missing a H1 on a page and multiple H1's on a page.

emerham committed:
Started working on some checks for missing H1's on a page

emerham committed:
Add missing and multiple H1 level checks in Accessibility

An update to the Accessibility tool has been made to check and notify the user if there are missing or multiple H1 level headings on their page. The code includes the user notifications and also pushes the results so the user can take immediate action. The aim is to ensure good SEO practices, accessibility, and content organization.

emerham committed:
Update multiple level 1 headings content

The content advising against the use of multiple level 1 headings on a webpage has been updated. The revised text provides additional details, including a directive to change other level 1 headings to heading 2 and links for learning more about proper heading structure for webpages. Also, it emphasizes the importance of a single, clear, descriptive level 1 heading for site accessibility, SEO ranking, usability, and clarity.

emerham committed:
Updated level 1 heading guidance in accessibility tips

The guidance messages for the missing level 1 heading in both the source and distribution files have been updated. The new messages provide more detailed instructions on how to add a level 1 heading using the Layout Builder. Also, the guidelines now handle situations where the Page Title may not serve as the preferred level 1 heading for the user.

Ruefa commented on dist/madrone-editoria11y.js:
```diff
@@ -59,6 +62,34 @@ document.addEventListener('ed11yRunCustomTests', () => {
+    title: 'Multiple Heading Level 1',
+    tip: () => `
+<p>You have <strong>multiple level 1 headings(h1)</strong> on your page. Typically, your page title is the level 1 heading of a page. You need to change the other level 1 headings to heading 2. You can learn more about heading structure for webpages on the <a href="https://drupal.oregonstate.edu/site-building-guide/elements">Site Building Guide</a> or <a href ="https://accessibility.oregonstate.e ...
+<p>This can cause confusion for your site visitors, particularly for those using assistive technologies like screen readers. The  level 1 heading is denote what your entire webpage is about.</p>
```
The level 1 heading is >**to**< denote. This sentence also has an extra space after `The`

emerham replied:
Should be fixed now

Ruefa reviewed (changes requested):
words

emerham committed:
Add checking for front page in madrone theme

Added a new variable to check if the current page is the front page in the madrone theme's template file. If there's an exception, the variable will be set to FALSE to ensure graceful degradation. This update aids in enabling custom functionality depending on the type of the page.

emerham committed:
Update styling for site name in madrone.css

The commit introduces a new class selector for site name with font styling in the madrone.css file. It sets the font family to "Open Sans", font size and line height to inherit, font weight to bold, and removes the bottom margin.

emerham committed:
Update site name handling in branding block template

Improved the way site names are handled within the system branding block template by introducing flexibility. The group name link now correctly references the content, and the site nameis properly wrapped with h1 tags when it's on the front page, improving the site's SEO.

emerham committed:
Fix syntax errors and improve accessibility checks in madrone-editoria11y.js

This commit corrects minor syntax errors and clarifies certain sentences to improve code readability. It also enhances the accessibility checks for the presence and count of H1 headings. We've added new logic to ensure that the theme level h1 is taken into account to prevent false positives in the H1-related accessibility checks.
